app/streaming/session.cpp

```diff
@@ -275,8 +275,9 @@ int Session::getDecoderCapabilities(StreamingPreferences::VideoDecoderSelection
     return caps;
 }
 
-Session::Session(NvComputer* computer, NvApp& app)
-    : m_Computer(computer),
+Session::Session(NvComputer* computer, NvApp& app, StreamingPreferences *preferences)
+    : m_Preferences(preferences ? preferences : new StreamingPreferences(this)),
```

Comment:
We need to delete `m_Preferences` in the destructor if `preferences == nullptr`

Reply:
As mentioned above, `Session` is set as parent of the `StreamingPreferences` (passing `this` as argument). This makes the `Session` object take ownership of the preferences object, and the preferences object gets deleted once `Session` is deleted.